plugins/woocommerce/src/Utilities/CallbackUtil.php

```diff
@@ -0,0 +1,120 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Utilities;
+
+/**
+ * Utility class for working with WordPress hooks and callbacks.
+ *
+ * @since 10.4.0
+ */
+final class CallbackUtil {
+
+	/**
+	 * Get a stable signature for a callback that can be used for hashing.
+	 *
+	 * This method normalizes callbacks into consistent string representations,
+	 * regardless of changes in dynamic properties in callback instances.
+	 *
+	 * @param callable|mixed $callback A PHP callback.
+	 * @return string Normalized callback signature.
+	 *
+	 * @since 10.4.0
+	 */
+	public static function get_callback_signature( $callback ): string {
```

Comment:
Noting that for anonymous classes, similar to the prior situation with anonymous functions, it'll return a signature that changes every time.

I believe this is not a commonly used feature, but I just wanted to note it so we do a quick check, at least against the Marketplace.

Reply:
For consistency I have modified it to use a reflection based approach, as it was doing already for closures.